### PR TITLE
[handwired/macroboard] Add separate f401 and f411 configs…

### PR DESCRIPTION
**Pull request by Michal-Szczepaniak**
…  add audio for f411, use pwm driver for rgblight and audio, add via

## Description

I made changes and split it into two boards with 2 different mappings one on f401 and seconf on f411 also changed some timers to have pwm audio and rgblight. Also added VIA.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/macroboard/keymaps/via/config.h [hunk 1]
```diff
@@ -0,0 +1 @@
+#define DYNAMIC_KEYMAP_LAYER_COUNT 6
```
Comment (drashna) [inline keyboards/handwired/macroboard/keymaps/via/config.h:1]: this needs a license header and an include guard.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/macroboard/keymaps/via/config.h (hunk 1 above)
Comment (Michal-Szczepaniak) [reply keyboards/handwired/macroboard/keymaps/via/config.h:18]: done

File: keyboards/handwired/macroboard/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [9] = LAYOUT_ortho_5x6(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
+};
```
Comment (drashna) [inline keyboards/handwired/macroboard/keymaps/via/keymap.c:89]: Dynamic layout size set to 6, only 6 layers needed here. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by drashna** [comment]
This has some merge conflicts that need to be resolved.

**Comment by Michal-Szczepaniak** [comment]
> This has some merge conflicts that need to be resolved.

how did this even happen lol, fixed

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/macroboard/f401/rules.mk [hunk 3]
```diff
@@ -4,9 +4,6 @@ MCU = STM32F401
 # Bootloader selection
 BOOTLOADER = stm32-dfu
 
-# Build Options
-#   change yes to no to disable
-#
 BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```
Comment (fauxpark) [inline keyboards/handwired/macroboard/f401/rules.mk:7]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/handwired/macroboard/config.h [hunk 4]
```diff
@@ -82,56 +73,11 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
-//#define NO_ACTION_LAYER
-//#define NO_ACTION_TAPPING
-//#define NO_ACTION_ONESHOT
-
-/* Bootmagic Lite key configuration */
-//#define BOOTMAGIC_LITE_ROW 0
-//#define BOOTMAGIC_LITE_COLUMN 0
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
```
Comment (fauxpark) [inline keyboards/handwired/macroboard/config.h:83]:
```suggestion
```

File: keyboards/handwired/macroboard/f411/rules.mk [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# MCU name
+MCU = STM32F411
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```
Comment (fauxpark) [inline keyboards/handwired/macroboard/f411/rules.mk:7]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Michal-Szczepaniak** [comment]
@fauxpark corrected!

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**